File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/startstop/Task.java
package tasks.startstop;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Control;

@JavaTaskInfo
public class Task extends JavaTask {

	@InputPortInfo(name = "In", dataType = Control.class)
	public InputPort in;

	@Override public void runnerStart() {
		System.out.println("Task started");
	}
	
	@Override
	public void execute(int grp) {
		System.out.println("Task run");
	}

	@Override public void runnerStop() {
		System.out.println("Task stopped");
	}
}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/startstop/Task.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort

@PythonTaskInfo
class Task(PythonTask):

    def __init_ports__(self):
        self.in_ = InputPort(name="in")

    def runner_start(self):
        print('Task : start')

    def execute(self, grp):
        print('Task : execute')


    def runner_stop(self):
        print('Task : stop')
